src/NavierStokes.cpp

```diff
@@ -0,0 +1,1050 @@
+                  fe_face_values[velocity].value(i, q)) *
+                fe_face_values.JxW(q);
+            }
+          }
+        }
+      }
+    }
+
+    cell->get_dof_indices(dof_indices);
+
+    system_matrix.add(dof_indices, cell_matrix);
+    system_rhs.add(dof_indices, cell_rhs);
+    deltat_lumped_mass_inv.add(dof_indices, cell_lumped_mass_inv);
+  }
+
+  for (const auto &i : locally_owned_dofs)
+  {
+    deltat_lumped_mass_inv[i] = deltat / deltat_lumped_mass_inv[i];
+  }
+
+  system_matrix.compress(VectorOperation::add);
+  system_rhs.compress(VectorOperation::add);
+  deltat_lumped_mass_inv.compress(VectorOperation::add);
+
+  // Dirichlet boundary conditions.
+  {
+    std::map<types::global_dof_index, double> boundary_values;
+    std::map<types::boundary_id, const Function<dim> *> boundary_functions;
+    std::vector<bool> c_mask(dim, true);
+    c_mask.push_back(false);
+
+    // We interpolate first the inlet velocity condition alone, then the wall
+    // condition alone, so that the latter "win" over the former where the two
+    // boundaries touch.
+    inlet_velocity.set_time(time);
+    boundary_functions[3] = &inlet_velocity;
+    VectorTools::interpolate_boundary_values(
+      dof_handler, boundary_functions, boundary_values,
+      ComponentMask(c_mask)
+    );
+
+    boundary_functions.clear();
+    Functions::ZeroFunction<dim> zero_function(dim + 1);
+    boundary_functions[0] = &zero_function;
+    boundary_functions[2] = &zero_function;
+    boundary_functions[4] = &zero_function;
+
+    VectorTools::interpolate_boundary_values(
+      dof_handler, boundary_functions, boundary_values,
+      ComponentMask(c_mask)
+    );
+
+    MatrixTools::apply_boundary_values(
+      boundary_values, system_matrix, solution_owned, system_rhs, false
+    );
+  }
+}
+
+void 
+NavierStokes::set_re_number(int Re)
+{
+  pcout << "-----------------------------------" << std::endl;
+  double U = inlet_velocity.get_mean_vel();
+  nu = (U * Diameter) / Re;
+  pcout << "New reynolds number setted to " << Re << " with nu = " 
+    << nu <<  " ." << std::endl;
+  pcout << "-----------------------------------" << std::endl;
+}
+
+
+void
+NavierStokes::solve_time_step(std::ostream &oss)
+{
+  auto start_time = std::chrono::high_resolution_clock::now();
+  SolverControl solver_control(10000, 1e-6 * system_rhs.l2_norm());
+
+  SolverGMRES<TrilinosWrappers::MPI::BlockVector> solver(solver_control);
+
+  //PreconditionIdentity preconditioner;
+
+  
+  PreconditionASIMPLE preconditioner;
+  preconditioner.initialize(
+    system_matrix.block(0, 0),
+    system_matrix.block(1, 0),
+    system_matrix.block(0, 1),
+    solution_owned
+  );
+  
+  
+  /*
+  PreconditionAYosida preconditioner;
+  preconditioner.initialize(
+    system_matrix.block(0, 0),
+    system_matrix.block(1, 0),
+    system_matrix.block(0, 1),
+    deltat_lumped_mass_inv,
+    solution_owned
+  );
+  */
+  
+  auto end_time_prec = std::chrono::high_resolution_clock::now();
```

Comment:
`deal.II` offers the class `TimerOutput`, that provides profiling functionality with a much better interface.